File: test_resources/rust/struct_with_external_type_buffers.rs
impl IntoVMBuffers for Test {
    fn read_from_buffers(bytes_reader: &mut BytesReader) -> Self {
        Self {
            touch_x: bytes_reader.read_f32(),
            touch_y: bytes_reader.read_f32(),
            touch_status: TouchStatus::read_from_buffers(bytes_reader),
        }
    }

    fn write_to_buffers(&self, bytes_writer: &mut BytesWriter) {
        bytes_writer.write_f32(self.touch_x);
        bytes_writer.write_f32(self.touch_y);
        self.touch_status.write_to_buffers(bytes_writer);
    }

    fn skip_in_buffers(bytes_reader: &mut BytesReader, count: u64) {
        for _ in 0..count {
            bytes_reader.read_f32();
            bytes_reader.read_f32();
            TouchStatus::read_from_buffers(bytes_reader);
        }
    }
}

File: test_resources/dart/struct_with_external_type_buffers.dart
class TestEmplaceToBuffers implements EmplaceToBuffers<Test> {
  const TestEmplaceToBuffers()

  @override
  void read(BytesReader reader, Test model) {
    model.touchX = reader.readFloat();
    model.touchY = reader.readFloat();
    const TouchStatusEmplaceToBuffers().read(reader, model.touchStatus);
  }

  @override
  void write(BytesWriter writer, Test model) {
    writer.writeFloat(model.touchX);
    writer.writeFloat(model.touchY);
    const TouchStatusEmplaceToBuffers().write(writer, model.touchStatus);
  }

  @override
  void skip(BytesReader reader, int count) {
    const TouchStatusEmplaceToBuffers().skip(reader, count);

    for (int i = 0; i < count; i += 1) {
      reader.readFloat();
      reader.readFloat();
    }
  }
}

class TestIntoToBuffers implements IntoToBuffers<Test> {
  const TestIntoToBuffers()

  @override
  Test read(BytesReader reader) {
    final touchX = reader.readFloat();
    final touchY = reader.readFloat();
    final touchStatus = const TouchStatusIntoBuffers().read(reader);

    return Test(
      touchX: touchX;
      touchY: touchY;
      touchStatus: touchStatus;
    );
  }

  @override
  void write(BytesWriter writer, Test model) {
    writer.writeFloat(model.touchX);
    writer.writeFloat(model.touchY);
    const TouchStatusIntoBuffers().write(writer, model.touchStatus);
  }

  @override
  void skip(BytesReader reader, int count) {
    const TouchStatusIntoBuffers().skip(reader, count);

    for (int i = 0; i < count; i += 1) {
      reader.readFloat();
      reader.readFloat();
    }
  }
}
